spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, :type => :controller do
+  before(:each) do
+    request.env["devise.mapping"] = Devise.mappings[:user]
+  end
+
+  describe "#facebook" do
+    it "should successfully authenticate using Facebook authentication hash and display a success message in flash" do
+      request.env["omniauth.auth"] = FactoryGirl.create(:auth_hash, :facebook)
+      get :facebook
+      user = User.find_by(email: 'testuser@facebook.com')
+      expect(user).not_to eq(nil)
+      expect(flash[:notice]).to eq("Successfully authenticated from Facebook account.")
+    end
+
+    it "should redirect the user to the new user registration form" do
```

Comment:
Good catch, I must have missed that when I removed context sections and refactored. I'll update that now.